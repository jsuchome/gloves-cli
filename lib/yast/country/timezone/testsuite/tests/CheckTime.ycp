/**
 * tests for Timezone::GetDateTime
 * Author: jsuchome@suse.cz
 * $Id: CheckTime.ycp 27936 2006-02-13 20:01:14Z olh $
 */
{

    // testedfiles: Timezone.ycp

    import "Testsuite";

    map READ = $[
	"sysconfig" : $[
	    "clock" : $[
		"TIMEZONE"	: "Europe/Prague",
		"HWCLOCK"	: "-u",
	    ],
	],
	"target" : $[
	    "size"		: 1,
	    "yast2"		: $[],
	],
    ];
    map EXECUTE	= $[
	"target" : $[
	    "bash_output"	: $[],
	],
    ];

    Testsuite::Init([READ,$[],EXECUTE], nil);

    import "Timezone";

    Testsuite::Test (``(Timezone::CheckTime ("1", "2", "3")), [], []);

    // wrong time:
    Testsuite::Test (``(Timezone::CheckTime ("24", "2", "3")), [], []);

    Testsuite::Test (``(Timezone::CheckTime ("1", "62", "3")), [], []);

    Testsuite::Test (``(Timezone::CheckTime ("1", "2", "63")), [], []);

    // wrong input:
    Testsuite::Test (``(Timezone::CheckTime ("", "2", "3")), [], []);

    Testsuite::Test (``(Timezone::CheckTime ("1", "2", "blah")), [], []);
}
