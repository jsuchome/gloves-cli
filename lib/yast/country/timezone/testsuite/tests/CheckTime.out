Return	true
Return	false
Return	false
Return	false
Return	false
Return	false
